platform/api/packet/api.go

```diff
@@ -478,22 +489,46 @@ boot`, a.opts.InstallerImageKernelURL, userdataURL, linuxConsole[a.opts.Board],
 }
 
 // device creation seems a bit flaky, so try a few times
-func (a *API) createDevice(hostname, ipxeScriptURL string) (device *packngo.Device, err error) {
+func (a *API) createDevice(hostname, ipxeScriptURL, id string) (*packngo.Device, error) {
 	for tries := apiRetries; tries >= 0; tries-- {
-		var response *packngo.Response
-		device, response, err = a.c.Devices.Create(&packngo.DeviceCreateRequest{
-			ProjectID:     a.opts.Project,
-			Facility:      []string{a.opts.Facility},
-			Plan:          a.opts.Plan,
-			BillingCycle:  "hourly",
-			Hostname:      hostname,
-			OS:            "custom_ipxe",
-			IPXEScriptURL: ipxeScriptURL,
-			Tags:          []string{"mantle"},
-		})
+		var (
+			device   *packngo.Device
+			response *packngo.Response
+			err      error
+		)
+
+		if id != "" {
+			plog.Infof("Recycling instance: %s", id)
+			// we update the device and we force a PXE boot in order to fetch the
+			// new configuration.
+			alwaysPXE := true
+			device, response, err = a.c.Devices.Update(id, &packngo.DeviceUpdateRequest{
+				AlwaysPXE:     &alwaysPXE,
+				IPXEScriptURL: &ipxeScriptURL,
+			})
```

Comment:
```suggestion
			})
			if err != nil {
				return nil, err
			}
```

Reply:
`return` at this point will make leave the `createDevice` method direct and we will have some remaining instances into EM. They won't be deleted when the flight will leave since the instance would have been consumed from the `devicesPool` channel.

I think we can keep in this way, if we fail to `update` and / or `reboot` the existing device, we just loop over. However, we could provide a bit more of context into the error:
```suggestion
			})
			if err != nil {
			    err = errors.New("updating instance: %w", err)
			}
```

What do you think ?

Reply:
ah, right, but to loop over you need to add a `continue` here

Reply:
otherwise later `a.DeleteDevice(device.ID)` gets called